File: src/main/java/com/nickcoblentz/montoya/settings/StringExtensionSetting.java
package com.nickcoblentz.montoya.settings;

import burp.api.montoya.MontoyaApi;

public class StringExtensionSetting extends GenericExtensionSetting<String> {

    public StringExtensionSetting(MontoyaApi api, String name, String storageKey, String defaultValue, ExtensionSettingSaveLocation defaultSaveLocation) {
        super(api, name, storageKey, defaultValue, defaultSaveLocation);
    }

    @Override
    public void loadFromProject() {
      setCurrentValue(getApi().persistence().extensionData().getString(getStorageKey()));
    }

    @Override
    public void loadFromPreferences() {
        setCurrentValue(getApi().persistence().preferences().getString(getStorageKey()));
    }

    @Override
    public String parseString(String value) {
        return value;
    }

    @Override
    public void saveToProject() {
        getApi().persistence().extensionData().setString(getStorageKey(),getCurrentValue());
    }

    @Override
    public void saveToPreferences() {
        getApi().persistence().preferences().setString(getStorageKey(),getCurrentValue());
    }
}

File: src/main/java/com/nickcoblentz/montoya/settings/StringExtensionSetting.kt
package com.nickcoblentz.montoya.settings

import burp.api.montoya.MontoyaApi

class StringExtensionSetting(
    api: MontoyaApi,
    name: String,
    storageKey: String,
    defaultValue: String,
    defaultSaveLocation: ExtensionSettingSaveLocation
) :
    GenericExtensionSetting<String>(api, name!!, storageKey, defaultValue, defaultSaveLocation) {
    override fun loadFromProject() {
        currentValue = api.persistence().extensionData().getString(storageKey)
    }

    override fun loadFromPreferences() {
        currentValue = api.persistence().preferences().getString(storageKey)
    }

    override fun parseString(value: String): String {
        return value
    }

    override fun setCurrentValue(value: String?) {
        if(value!=null)
            currentValue=value
    }

    override fun saveToProject() {
        api.persistence().extensionData().setString(storageKey, currentValue)
    }

    override fun saveToPreferences() {
        api.persistence().preferences().setString(storageKey, currentValue)
    }
}